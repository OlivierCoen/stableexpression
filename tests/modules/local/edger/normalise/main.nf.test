nextflow_process {

    name "Test Process EDGER_NORMALISE"
    script "modules/local/edger/normalise/main.nf"
    process "EDGER_NORMALISE"
    tag "edger"
    tag "normalisation"
    tag "module"

    test("Very small dataset") {

        when {

            process {
                """
                meta = [accession: "accession", design: file('$projectDir/tests/input/normalise/design.csv')]
                input[0] = [meta, file('$projectDir/tests/input/normalise/all_counts.csv')]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

}
