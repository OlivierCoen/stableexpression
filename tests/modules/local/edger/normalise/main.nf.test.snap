{
    "Very small dataset - no zeros": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T12:40:17.115538"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "accession",
                            "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                        },
                        "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                    ]
                ],
                "1": [
                    [
                        "EDGER_NORMALISE",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "2": [
                    [
                        "EDGER_NORMALISE",
                        "edgeR",
                        "4.0.16"
                    ]
                ],
                "csv": [
                    [
                        {
                            "accession": "accession",
                            "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                        },
                        "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-29T12:36:28.080031"
    },
    "Very small dataset - with zeros": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T12:39:11.877808"
    }
}
